src/construction.cpp

```diff
@@ -105,7 +105,7 @@ std::vector<construction *> constructions_by_desc( const std::string &descriptio
 {
     std::vector<construction *> result;
     for( auto &constructions_a : constructions ) {
-        if( constructions_a.description == description ) {
+        if( constructions_a.desc() == description ) {
```

Comment:
It seems that extensively calling translation routines for constructions causes major slowdown during construction menu navigation. We should probably use some kind of id instead of string compare to find available construction stages for certain construction.

Reply:
So we basically have `std::vector<std::string> constructs;` which contains descriptions of all constructions (109 in `dda` mod).

After construction selection is changed in the menu, we need to populate construction stages to make it possible to display their descriptions in separate window without navigating menu.

It is done via `std::vector<construction *> options = constructions_by_desc( current_desc );`.

Here `current_desc` is translated string, so it is being compared with other translated construction descriptions for all constructions and translation overhead adds up.

I guess we need to utilize some kind of id for construction and its stages, so we don't have to compare strings to get all stages of certain construction.

Reply:
I see. Sorry about this issue. There were untranslated descriptions comparisons prior my change. So I guess we may revert this PR to quick-fix the issue.